db/migrate/20221202130039_fill_project_journals_with_existing_data.rb

```diff
@@ -0,0 +1,95 @@
+class FillProjectJournalsWithExistingData < ActiveRecord::Migration[7.0]
+  def up
+    create_journal_entries_for_projects
+  end
+
+  def down
+    delete_journal_entries_for_projects
+  end
+
+  def create_journal_entries_for_projects
+    sql = <<~SQL.squish
+      WITH project_journals_insertion AS (
+        INSERT INTO project_journals(
+            name,
+            description,
+            public,
+            parent_id,
+            identifier,
+            active,
+            templated
+          )
+        SELECT name,
+          description,
+          public,
+          parent_id,
+          identifier,
+          active,
+          templated
+        FROM projects
+        RETURNING id,
+          identifier
+      ),
+      journals_insertion AS (
+        INSERT into journals (
+            journable_id,
+            journable_type,
+            user_id,
+            created_at,
+            version,
+            activity_type,
+            data_type,
+            data_id
+          )
+        SELECT projects.id,
+          'Project',
+          :user_id,
+          projects.created_at,
+          1,
+          'project_attributes',
+          'Journal::ProjectJournal',
+          project_journals_insertion.id
+        FROM projects
+          LEFT JOIN project_journals_insertion ON projects.identifier = project_journals_insertion.identifier
+        RETURNING id,
+          journable_id
+      ),
+      customizable_journals_insertion AS (
+        INSERT into customizable_journals (
+            journal_id,
+            custom_field_id,
+            value
+          )
+        SELECT journals_insertion.id,
+          custom_field_id,
+          value
+        FROM custom_values
+          LEFT JOIN journals_insertion ON custom_values.customized_id = journals_insertion.journable_id
+        WHERE custom_values.customized_type = 'Project'
+      )
+      SELECT COUNT(1)
+      FROM journals_insertion;
+    SQL
+    sql = ::ActiveRecord::Base.sanitize_sql_array([sql, { user_id: User.system.id }])
+    ::ActiveRecord::Base.connection.execute(sql)
```

Comment:
Since the code is executed in a migration, you should be able to shorten this call to just `execute(sql)` same as you have done below in the `delete` method.